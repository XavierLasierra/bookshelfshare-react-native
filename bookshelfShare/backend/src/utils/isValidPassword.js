const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
