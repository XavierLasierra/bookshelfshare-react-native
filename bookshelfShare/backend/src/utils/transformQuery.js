function transformQuery(query) {
  let transformedQuery = {};
  const entries = Object.entries(query);
  if (entries.length > 0) {
    entries.forEach(([property, filter]) => {
      if (property.includes('_')) {
        const [propertyName, symbol] = property.split('_');
        transformedQuery = { ...transformedQuery, [propertyName]: { [`$${symbol}`]: filter } };
      } else {
        transformedQuery = { ...transformedQuery, [property]: { $regex: filter, $options: 'i' } };
      }
    });
  }
  return transformedQuery;
}

module.exports = { transformQuery };
