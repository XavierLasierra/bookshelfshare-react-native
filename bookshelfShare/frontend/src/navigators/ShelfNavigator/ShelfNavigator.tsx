import React from 'react';
import { createStackNavigator, CardStyleInterpolators } from '@react-navigation/stack';

import UserShelfList from '../../components/UserShelfList/UserShelfList';
import NewShelf from '../../components/NewShelf/NewShelf';
import ShelfDetail from '../../components/ShelfDetail/ShelfDetail';
import BookDetail from '../../components/BookDetail/BookDetail';

export default function ShelfNavigator() {
  const Stack = createStackNavigator();
  return (
    <Stack.Navigator
      initialRouteName="UserShelfList"
      screenOptions={{
        headerShown: false,
        cardShadowEnabled: false,
        cardOverlayEnabled: false,
        cardStyleInterpolator: CardStyleInterpolators.forFadeFromCenter
      }}
    >
      <Stack.Screen
        name="UserShelfList"
        component={UserShelfList}
      />
      <Stack.Screen
        name="NewShelf"
        component={NewShelf}
      />
      <Stack.Screen
        name="ShelfDetail"
        component={ShelfDetail}
      />
      <Stack.Screen
        name="BookDetail"
        component={BookDetail}
      />
    </Stack.Navigator>
  );
}
