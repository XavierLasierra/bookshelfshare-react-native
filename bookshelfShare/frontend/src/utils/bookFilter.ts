export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
