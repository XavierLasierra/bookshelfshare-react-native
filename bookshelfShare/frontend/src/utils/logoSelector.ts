import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
