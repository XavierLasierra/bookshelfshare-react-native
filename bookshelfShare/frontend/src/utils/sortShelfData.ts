export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
