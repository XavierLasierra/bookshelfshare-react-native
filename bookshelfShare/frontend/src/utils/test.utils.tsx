import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
