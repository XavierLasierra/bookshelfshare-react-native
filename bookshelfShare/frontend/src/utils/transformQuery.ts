interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
