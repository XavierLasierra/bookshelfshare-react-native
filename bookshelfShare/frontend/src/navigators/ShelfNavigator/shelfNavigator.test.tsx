import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import ShelfNavigator from './ShelfNavigator';
import { render } from '../../utils/test.utils';
import loggedUserMock from '../../mocks/loggedUser.mock';

jest.useFakeTimers();

describe('Given a ShelfNavigator component', () => {
  describe('When it is rendered', () => {
    test('Then should match the snapshot', () => {
      const initialState = {
        loggedUser: { userData: loggedUserMock.user }
      };
      const screen = render(
        <NavigationContainer>
          <ShelfNavigator />
        </NavigationContainer>,
        initialState
      );
      expect(screen).toMatchSnapshot();
    });
  });
});
