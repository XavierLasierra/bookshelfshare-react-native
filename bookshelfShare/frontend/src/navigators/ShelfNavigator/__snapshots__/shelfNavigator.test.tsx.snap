// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ShelfNavigator component When it is rendered Then should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                enableTrackpadTwoFingerGesture={true}
                enabled={false}
                hitSlop={
                  Object {
                    "right": -700,
                  }
                }
                maxDeltaY={20}
                minOffsetX={5}
                onHandlerStateChange={[Function]}
              >
                <View
                  needsOffscreenAlphaCompositing={true}
                  style={
                    Object {
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "backgroundColor": "rgb(242, 242, 242)",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <RCTSafeAreaView
                          emulateUnlessSupported={true}
                          style={
                            Object {
                              "backgroundColor": "#F5F5F5",
                              "minHeight": "100%",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "flex-end",
                                "backgroundColor": "#F5F5F5",
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "#D9D8C4",
                                  "borderBottomLeftRadius": 25,
                                  "justifyContent": "center",
                                  "padding": 10,
                                  "width": "100%",
                                }
                              }
                            >
                              <IconMock
                                height={45}
                                width={45}
                              />
                            </View>
                            <View
                              style={
                                Object {
                                  "height": 0,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "backgroundColor": "#D9D8C4",
                                    "height": 20,
                                    "width": 20,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "#F5F5F5",
                                      "borderTopRightRadius": 25,
                                      "height": 20,
                                      "width": 20,
                                    }
                                  }
                                />
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "paddingHorizontal": 30,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "justifyContent": "space-between",
                                  "paddingVertical": 10,
                                  "width": "100%",
                                }
                              }
                            >
                              <TextInput
                                maxLength={25}
                                placeholder="Search in all your shelves"
                                style={
                                  Object {
                                    "backgroundColor": "#FFFFFF",
                                    "borderColor": "#ECECEC",
                                    "borderRadius": 10,
                                    "borderWidth": 1,
                                    "fontSize": 16,
                                    "height": "100%",
                                    "paddingHorizontal": 20,
                                    "width": "85%",
                                  }
                                }
                              />
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={false}
                                nativeID="animatedComponent"
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "opacity": 1,
                                  }
                                }
                              >
                                <IconMock
                                  height={35}
                                  width={35}
                                />
                              </View>
                            </View>
                            <RCTScrollView
                              ListFooterComponent={
                                <TouchableOpacity
                                  onPress={[Function]}
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "justifyContent": "center",
                                      "marginBottom": 450,
                                      "padding": 30,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "alignItems": "center",
                                          "backgroundColor": "#FFFFFF",
                                          "borderRadius": 100,
                                          "elevation": 5,
                                          "height": 60,
                                          "justifyContent": "center",
                                          "width": 60,
                                        },
                                        Object {
                                          "backgroundColor": "#ECECEC",
                                          "height": 40,
                                          "marginRight": 20,
                                          "width": 40,
                                        },
                                      ]
                                    }
                                  >
                                    <Text
                                      style={
                                        Object {
                                          "fontSize": 20,
                                        }
                                      }
                                    >
                                      +
                                    </Text>
                                  </View>
                                  <Text
                                    style={
                                      Object {
                                        "color": "#3E3C37",
                                        "fontSize": 16,
                                      }
                                    }
                                  >
                                    Create new bookshelf
                                  </Text>
                                </TouchableOpacity>
                              }
                              data={Array []}
                              getItem={[Function]}
                              getItemCount={[Function]}
                              keyExtractor={[Function]}
                              onContentSizeChange={[Function]}
                              onLayout={[Function]}
                              onMomentumScrollBegin={[Function]}
                              onMomentumScrollEnd={[Function]}
                              onScroll={[Function]}
                              onScrollBeginDrag={[Function]}
                              onScrollEndDrag={[Function]}
                              removeClippedSubviews={false}
                              renderItem={[Function]}
                              scrollEventThrottle={50}
                              showsVerticalScrollIndicator={false}
                              stickyHeaderIndices={Array []}
                              viewabilityConfigCallbackPairs={Array []}
                            >
                              <View>
                                <View
                                  onLayout={[Function]}
                                >
                                  <View
                                    accessible={true}
                                    collapsable={false}
                                    focusable={true}
                                    nativeID="animatedComponent"
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "justifyContent": "center",
                                        "marginBottom": 450,
                                        "opacity": 1,
                                        "padding": 30,
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Array [
                                          Object {
                                            "alignItems": "center",
                                            "backgroundColor": "#FFFFFF",
                                            "borderRadius": 100,
                                            "elevation": 5,
                                            "height": 60,
                                            "justifyContent": "center",
                                            "width": 60,
                                          },
                                          Object {
                                            "backgroundColor": "#ECECEC",
                                            "height": 40,
                                            "marginRight": 20,
                                            "width": 40,
                                          },
                                        ]
                                      }
                                    >
                                      <Text
                                        style={
                                          Object {
                                            "fontSize": 20,
                                          }
                                        }
                                      >
                                        +
                                      </Text>
                                    </View>
                                    <Text
                                      style={
                                        Object {
                                          "color": "#3E3C37",
                                          "fontSize": 16,
                                        }
                                      }
                                    >
                                      Create new bookshelf
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </RCTScrollView>
                          </View>
                          <View
                            style={
                              Object {
                                "height": 130,
                                "width": "100%",
                              }
                            }
                          />
                        </RCTSafeAreaView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
